src/app/parcels/form/ParcelForm.tsx

```diff
@@ -178,10 +197,25 @@ const ParcelForm: React.FC<ParcelFormProps> = ({
         setSubmitDisabled(false);
     };
 
+    // @ts-ignore
     return (
         <CenterComponent>
             <StyledForm>
                 <Title>Parcel Form</Title>
+                <Button
+                    variant="contained"
+                    type="button"
+                    onClick={() => {
+                        if (clientId) {
+                            getExpandedClientDetails(clientId).then((data) => {
+                                setActiveData(data);
+                                setModalIsOpen(true);
+                            });
```

Comment:
Sorry I wasn't clear before:

Seems like we call `getExpandedClientDetails` which gets all the client details twice. My original suggestion was to create a function that just gets the full name of the client first, and when needed, we can separately retrieve the rest.

However, I'm not sure how much of a performance gain that is, considering we would be selecting less fields from a row in the table. So I think the best way to tackle this would be to:
- have a state `clientDetails` (which I think should  replace `activeData`)
- call `getExpandedClientDetails` in useEffect (as we currently do) and store the response into `clientDetails`
- when the client details button is clicked, simply do `setModalIsOpen(true)` to open the modal